File: Algorithms/Python/special_diamond_pattern.py
# Special diamond pattern with following conditions
        # Single For Loop
        # Single if else if ladder
        # Only 3 variables in whole program
num = int(input())
row_controller = 1 - num
i = 1

while row_controller < num:
    if i < abs(row_controller) + 1:
        print(" " , end = "")
        i += 1
    elif i < 2 * num - (1 + abs(row_controller)):
        print( "*" , end = "")
        i += 1
    else:
        print("*")
        i = 1
        row_controller += 1

File: Algorithms/C/special_diamond_pattern.c
// Special diamond pattern with following conditions
        // Single For Loop
        // Single if else if ladder
        // Only 3 variables in whole program
#include<stdio.h>
#include<stdlib.h>
int main()
{
    int i , num , row_controller;
    scanf("%d" , &num);

    row_controller = 1 - num;

    for(i = 1 ; row_controller < num ; i++)
{
        if(i < 1 + abs(row_controller))
            printf(" ");
        else if(i < 2 * num - (1 + abs(row_controller)))
            printf("*" , abs(row_controller) + 1);
        else
        {
            printf("*\n" , abs(row_controller) + 1);
            i = 0;
            row_controller += 1;
        }
    }
    return 0;
}